opm/simulators/wells/StandardWellPrimaryVariables.cpp

```diff
@@ -438,45 +444,31 @@ volumeFraction(const unsigned compIdx) const
         return EvalWell(numWellEq_ + Indices::numEq, 1.0);
     }
 
+    if (has_gfrac_variable && compIdx == Indices::canonicalToActiveComponentIndex(FluidSystem::gasCompIdx)) {
+        return evaluation_[GFrac];
+    }
+
+    if (Indices::enableSolvent && compIdx == (unsigned)Indices::contiSolventEqIdx) {
+        return evaluation_[SFrac];
+    }
+
     if (FluidSystem::phaseIsActive(FluidSystem::oilPhaseIdx)) {
         if (has_wfrac_variable && compIdx == Indices::canonicalToActiveComponentIndex(FluidSystem::waterCompIdx)) {
             return evaluation_[WFrac];
         }
-
-        if (has_gfrac_variable && compIdx == Indices::canonicalToActiveComponentIndex(FluidSystem::gasCompIdx)) {
-            return evaluation_[GFrac];
-        }
-
-        if (Indices::enableSolvent && compIdx == (unsigned)Indices::contiSolventEqIdx) {
-            return evaluation_[SFrac];
-        }
-    }
-    else if (FluidSystem::phaseIsActive(FluidSystem::waterPhaseIdx)) {
-        if (has_gfrac_variable && compIdx == Indices::canonicalToActiveComponentIndex(FluidSystem::gasCompIdx)) {
-            return evaluation_[GFrac];
-        }
     }
 
     // Oil or WATER fraction
     EvalWell well_fraction(numWellEq_ + Indices::numEq, 1.0);
-    if (FluidSystem::phaseIsActive(FluidSystem::oilPhaseIdx)) {
-        if (FluidSystem::phaseIsActive(FluidSystem::waterPhaseIdx)) {
-            well_fraction -= evaluation_[WFrac];
-        }
-
-        if (FluidSystem::phaseIsActive(FluidSystem::gasPhaseIdx)) {
-            well_fraction -= evaluation_[GFrac];
-        }
-
-        if constexpr (Indices::enableSolvent) {
-            well_fraction -= evaluation_[SFrac];
-        }
+    if (has_wfrac_variable && FluidSystem::phaseIsActive(FluidSystem::oilPhaseIdx)) {
```

Comment:
can you modify the comment `// Oil or WATER fraction` to be more specific. 

Basically, it will be oil fraction when oil phase is active and water fraction when oil phase is not active.